media/js/base/mozilla-fxa-link.js

```diff
@@ -34,6 +34,49 @@ if (typeof window.Mozilla === 'undefined') {
         return href;
     };
 
+    /**
+     * Intercept event handler for FxA links and forms, lets the browser drive the FxA Flow using
+     * the `showFirefoxAccounts` UITour API. Attaches several UTM parameters from the current page
+     * that will be forwarded to the browser and later on to FxA services.
+     * @param event {Event}
+     * @private
+     */
+    FxaLink.interceptFxANavigation = function(event) {
+        event.preventDefault();
+        var urlParams = new window._SearchParams().utmParams();
+        var attributionData = Mozilla.UtmUrl.getAttributionData(urlParams);
+        var extraURLParams = {};
+        if (attributionData) {
+            var utms = ['utm_source', 'utm_campaign', 'utm_content', 'utm_term', 'utm_medium'];
+            for (var i = 0; i < utms.length; i++) {
+                if (Object.prototype.hasOwnProperty.call(attributionData, utms[i])) {
+                    extraURLParams[utms[i]] = attributionData[utms[i]];
+                }
+            }
+        }
+
+        var email = document.getElementById('fxa-email-field');
+        if (email) {
+            email = email.value;
+        }
+        var formElem = document.getElementById('fxa-email-form');
+        if (formElem) {
+            var deviceId = formElem.querySelector('[name="device_id"]');
+            var flowId = formElem.querySelector('[name="flow_id"]');
+            var flowBeginTime = formElem.querySelector('[name="flow_begin_time"]');
+            if (deviceId && deviceId.value) {
+                extraURLParams['device_id'] = deviceId.value;
+            }
+            if (flowId && flowId.value) {
+                extraURLParams['flow_id'] = flowId.value;
+            }
+            if (flowBeginTime && flowBeginTime.value) {
+                extraURLParams['flow_begin_time'] = parseInt(flowBeginTime.value, 10);
+            }
+        }
```

Comment:
I don't think we need this logic that deals with the form, as we split this out into `mozilla-fxa-form.js`, right?